File: WoofIOS/SwiftPackages/Presentation/Profile/Tests/PresentationProfileTests/Navigation/FakeNavigator.swift
import PresentationNavigation

final class FakeNavigator: NavigatorImplementing {
    private(set) var destinations: [WoofDestination] = []
    private(set) var backCount = 0

    func goTo(_ destination: WoofDestination) {
        destinations.append(destination)
    }

    func back() {
        backCount += 1
    }
}

File: WoofAndroid/presentation/profile/src/test/kotlin/com/perrystreet/woof/presentation/profile/navigation/FakeNavigator.kt
package com.perrystreet.woof.presentation.profile.navigation

import com.perrystreet.woof.presentation.navigation.INavigator
import com.perrystreet.woof.presentation.navigation.WoofDestination

class FakeNavigator : INavigator {
    val destinations = mutableListOf<WoofDestination>()
    var backCount: Int = 0

    override fun goTo(destination: WoofDestination) {
        destinations += destination
    }

    override fun back() {
        backCount++
    }
}
